string_manipulation.rb

```diff
@@ -130,6 +130,52 @@ def encode_repeating(my_string)
   return my_string
 end
 
+def encode_repeating2(my_string)
+  return my_string if my_string.length < 3
+
+  entering = 0
+  reading = 0
+  count = 1
+
+  while reading < my_string.length - 1
+
+    if my_string[reading] == my_string[reading+1] #if the next two are the same
+      count += 1
+      reading += 1
+    elsif count > 2 #they arent the same but the count is enough to replace
+      my_string[entering] = my_string[reading]
+      my_string[entering + 1]= "#{count.to_s}"
+      count = 1
+      entering += 2
+      reading += 1
+    elsif count == 2 #they arent the same and the count is 2
+      count = 1

```

Comment:
You will still need to copy from reading to entering two times in this case. e.g. "aaaabbccc" will become "a4aabbcc" When b is counted to 2, entering index will be at 2 and reading index will be 5. You want to copy over my_string[5] to my_string[2] to make it "a4babbcc" and then copy over my_string[5] one more time to my_string[3] to make it "a4bbbbcc".